cpp/src/parquet/metadata.cc

```diff
@@ -623,6 +623,11 @@ bool ApplicationVersion::HasCorrectStatistics(Type::type col_type,
     return false;
   }
 
+  // Null only arrays do not have proper statistics
+  if (statistics.null_count > 0 && statistics.distinct_count == 0) {
```

Comment:
We never actually compute distinct_count so it's always going to be zero I think